cluster_experiments/perturbator.py

```diff
@@ -350,6 +351,266 @@ def _sample_beta_effect(
         return np.random.beta(a, b, n)
 
 
+class BetaRelativePerturbator(NormalPerturbator, RelativePositivePerturbator):
+    """
+    A stochastic Perturbator for continuous targets that applies a  sampled
+    effect from a scaled Beta distribution. It applies the effect multiplicatively.
+
+    The average effect is defined for values in the specified range (range_min, range_max).
+    It's recommended to set range_min and range_max in a "symmetric way" around 0,
+    such that log(1 + range_min) = -log(range_max). This ensures to have an
+    equal range of perturbations that relatively decrease the target as
+    perturbations that relatively increase the target.
+
+    The number of samples with 0 as target remains unchanged.
+
+    The stochastic effect is sampled from a beta distribution with parameters mean and
+    variance, which is scaled to the range (range_min, range_max).
+    If variance is not provided, the variance is abs(mean).
+
+    ```
+    target -> target * (1 + effect), where effect ~ Beta(a, b)
```

Comment:
It's really just a linear transformation that's applied to the usual beta distribution with support in [0, 1], so I think there's really nothing magical going on here^^
Not sure if there's a small typo in your comment and you meant to say to add a Demo, in which case yes, this would be a good idea. I never looked at the other Demos though and my doubt right now is that considering that this perturbator could still change a bit once I try to integrate it into the experimentation-analysis workflow, there's a risk that the Demo might become outdated without us realizing. Or is "Desmos" some tool or package that I'm not aware of?

Reply:
I meant Desmos actually: https://www.desmos.com/calculator/frnbtrv4bq